mono/mini/debugger-agent.c

```diff
@@ -2722,8 +2722,8 @@ compute_frame_info (MonoInternalThread *thread, DebuggerTlsData *tls)
 		/* Have to use the state saved by the signal handler */
 		process_frame (&tls->async_last_frame, NULL, &user_data);
 		mono_jit_walk_stack_from_ctx_in_thread (process_frame, tls->domain, &tls->async_ctx, FALSE, thread, tls->async_lmf, &user_data);
-	} else if (tls->has_context) {
-		mono_jit_walk_stack_from_ctx_in_thread (process_frame, tls->domain, &tls->ctx, FALSE, thread, tls->lmf, &user_data);
+	} else if (has_context) {
+		mono_jit_walk_stack_from_ctx_in_thread (process_frame, tls->domain, context, FALSE, thread, lmf, &user_data);
```

Comment:
**compute_frame_info** below calls **compute_frame_info_with_context** with the same parameters are previously, so the logic for **compute_frame_info** is unchanged. 

**compute_frame_info_with_context** is otherwise only called from **compute_frame_count**, which always has a context and tls context is not used to any other called function, the tls values are used a "arguments" for this function.